backend/src/main.rs

```diff
@@ -23,19 +19,29 @@ async fn board_record_move(path: web::Path<(u32, String)>) -> impl Responder {
     let (board_id, board_move) = path.into_inner();
     println!("board_id: {}, move: {}", board_id, board_move);
     let game = "game.txt";
-    let history = History::from_filepath(game);
+    let mut history;
```

Comment:
you might consider something like this for the next couple `match` statements
```suggestion
    let mut history = History.from_filepath(game)
        .map_err(|e| HttpResponse::BadRequest().body(format!("{}", e)))?;
```